tests/unit/unsupervised_inference_evidence_lm_test.py

```diff
@@ -0,0 +1,245 @@
+# Copyright 2025 Thousand Brains Project
+#
+# Copyright may exist in Contributors' modifications
+# and/or contributions to the work.
+#
+# Use of this source code is governed by the MIT
+# license that can be found in the LICENSE file or at
+# https://opensource.org/licenses/MIT.
+
+
+import copy
+import shutil
+import tempfile
+import unittest
+from pprint import pprint
+
+import numpy as np
+
+from tbp.monty.frameworks.config_utils.config_args import (
+    LoggingConfig,
+    MontyFeatureGraphArgs,
+    PatchAndViewMontyConfig,
+)
+from tbp.monty.frameworks.config_utils.make_dataset_configs import (
+    EnvironmentDataLoaderPerObjectEvalArgs,
+    EnvironmentDataLoaderPerObjectTrainArgs,
+    ExperimentArgs,
+    PredefinedObjectInitializer,
+)
+from tbp.monty.frameworks.environments import embodied_data as ED
+from tbp.monty.frameworks.experiments import MontyObjectRecognitionExperiment
+from tbp.monty.frameworks.models.evidence_matching import (
+    EvidenceGraphLM,
+    MontyForEvidenceGraphMatching,
+)
+from tbp.monty.frameworks.models.evidence_unsupervised_inference_matching import (
+    MontyForUnsupervisedEvidenceGraphMatching,
+)
+from tbp.monty.simulators.habitat.configs import (
+    EnvInitArgsPatchViewMount,
+    PatchViewFinderMountHabitatDatasetArgs,
+)
+from tests.unit.resources.unit_test_utils import BaseGraphTestCases
+
+
+class UnsInfEvidenceLMTest(BaseGraphTestCases.BaseGraphTest):
```

Comment:
I would write this out. It's not obvious what the abbreviation means and there is no harm in having a longer class name